server/depot.go

```diff
@@ -98,14 +112,93 @@ func (d fileDepot) Serial() (*big.Int, error) {
 		}
 		return s, nil
 	}
-	data, err := ioutil.ReadFile(name)
+	file, err := os.Open(name)
 	if err != nil {
 		return nil, err
 	}
-	serial := s.SetBytes(data)
+	defer file.Close()
+	r := bufio.NewReader(file)
+	// is there a better way??
+	data, err := r.ReadString('\r')
+	data = strings.TrimSuffix(data ,"\r")
+	data = strings.TrimSuffix(data ,"\n")
+	if err != nil && err != io.EOF {
+		return nil, err
+	}
+	serial, success := s.SetString(data,16)
+	if success != true  {
+		return nil, errors.New("Could not convert "+string(data)+" to serial number")
+	}
 	return serial, nil
 }
 
+func (d fileDepot) writeDB(cn string, serial *big.Int, filename string, cert *x509.Certificate) error {
+
+	var dn bytes.Buffer
+	var dbEntry bytes.Buffer
+
+	if err := os.MkdirAll(d.dirPath, 0755); err != nil {
+		return err
+	}
+	name := d.path("index.txt")
+
+	issuer := cert.Issuer
+
+	file, err := os.OpenFile(name, os.O_CREATE | os.O_RDWR | os.O_APPEND, dbPerm)
+	if err != nil {
+	    fmt.Printf("Could not append to "+name+" : %q\n", err.Error())
```

Comment:
If you'd like to annotate the error, consider using `fmt.Errorf` instead. 
https://golang.org/pkg/fmt/#Errorf